File: pset6/pset6/credit.py
from cs50 import get_int


def main():
    no = get_int("Enter your credit card no. :")
    lenght(no)  # function call


def lenght(n):  # to check card  is 13-16 no.or not
    if n < 1000000000000 or n > 10000000000000000:
        print("INVALID")
    else:
        checksum(n)  # function call

        
def checksum(n):    # to check whether it is valid or not or belongs to which card service company
    t = n
    su = 0
    sum1 = 0
    r = 0
    while n != 0:   # calculating sum and sum 1
        su += n % 10
        n = n // 10
        r = (n % 10) * 2
        if r >= 10:
            sum1 += r % 10
            r = r // 10
            sum1 += r % 10
            r = r // 10
            n = n // 10
        
        else:
            sum1 += r
            n = n // 10
    
  # print(su, sum1)
    usum = su + sum1
    print(usum)        # give the sum 
    usum = usum % 10  # checking usum last last digit is 0 or not
    if usum == 0:
        while t > 100:
            t = t // 10
        
        if t == 34 or t == 37:
            print("AMEX")   # check for AMEX card
        elif t >= 51 and t <= 55:
            print("MASTERCARD")  # check for MASTERCARD
        
        else:
            t = t // 10
            if t == 4:
                print("VISA")  # check for VISA card
            
            else:
                print("INVALID")   # case for an invalid type number with correct checksum
    else:
        print("INVALID")  # card is not vaild
 
        
main()

File: pset2/pset2/credit.c
#include <cs50.h>
#include <stdio.h>

void lenght(long n);      //function declartion
void checksum(long n);    //function declartion
int main(void)
{

    long no = get_long("Enter your credit card no. :");
    lenght(no);          // function call

}
void lenght(long n)                // to check card  is 13-16 no.or not
{
    if (n < 1000000000000 || n > 10000000000000000)
    {
        printf("INVALID\n");
    }
    else
    {
        checksum(n);    //function call
    }
}

void checksum(long n)   // to check whether it is valid or not or belongs to which card service company
{  
    long t = n;
    static int sum = 0;
    static int sum1 = 0;
    int r;
    while (n != 0)                              // calculating sum and sum 1
    {
        sum += n % 10;
        n = n / 10;

        r = (n % 10) * 2;
        if (r >= 10)
        {
            sum1 += r % 10;
            r = r / 10;
            sum1 += r % 10;
            r = r / 10;
            n = n / 10;
        }
        else
        {
            sum1 += r;
            n = n / 10;
        }
    }
    int usum = sum + sum1;   
    printf("%i",usum);//give the sum 
    usum = usum %  10;                                // checking usum last last digit is 0 or not
    if (usum == 0)
    {
        while (t > 100)
        {
            t /= 10;
        }
        if (t == 34 || t == 37)
        {
            printf("AMEX\n");                           //check for AMEX card
        }
        else if (t >= 51 && t <= 55)
        {
            printf("MASTERCARD\n");                     //check for MASTERCARD
        }
        else
        {
            t /= 10;
            if (t == 4)
            {
                printf("VISA\n");                       //check for VISA card
            }
            else
            {
                printf("INVALID\n");                    //case for an invalid type number with correct checksum
            }
        }
    }
    else 
    {
        printf("INVALID\n");          // card is not vaild
        
    }
   
}

